src_lockfree/michael_scott_queue_node.ml

```diff
@@ -3,10 +3,12 @@ module Atomic = Transparent_atomic
 type ('a, _) t =
   | Nil : ('a, [> `Nil ]) t
   | Next : {
-      next : ('a, [ `Nil | `Next ]) t Atomic.t;
+      mutable next : ('a, [ `Nil | `Next ]) t;
       mutable value : 'a;
     }
       -> ('a, [> `Next ]) t
 
-let[@inline] make value = Next { next = Atomic.make Nil; value }
-let[@inline] as_atomic (Next r : ('a, [ `Next ]) t) = r.next
+let[@inline] make value = Next { next = Nil; value }
+
+external as_atomic : ('a, [ `Next ]) t -> ('a, [ `Nil | `Next ]) t Atomic.t
```

Comment:
`next_as_atomic` seems like a better name for readability (and I think it is the name you usually use). Also, I won't be again a comment in the code about this line :)